<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1107</width>
    <height>676</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Simple Spring-Mass System</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="OpenGLWidget" name="openGLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>801</width>
      <height>601</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>70</y>
      <width>251</width>
      <height>421</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="4" column="1">
      <widget class="QLabel" name="label_2">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Spring constant</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QLCDNumber" name="lcd_mass">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="digitCount">
        <number>4</number>
       </property>
       <property name="segmentStyle">
        <enum>QLCDNumber::Flat</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="4">
      <widget class="QLabel" name="label_9">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>N/m</string>
       </property>
      </widget>
     </item>
     <item row="1" column="4">
      <widget class="QLabel" name="label_6">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>N/m</string>
       </property>
      </widget>
     </item>
     <item row="2" column="4">
      <widget class="QLabel" name="label_7">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>m</string>
       </property>
      </widget>
     </item>
     <item row="4" column="3">
      <widget class="QLCDNumber" name="lcd_spring_constant">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="digitCount">
        <number>4</number>
       </property>
       <property name="segmentStyle">
        <enum>QLCDNumber::Flat</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label_4">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Rest length</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLCDNumber" name="lcd_rest_length">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="smallDecimalPoint">
        <bool>false</bool>
       </property>
       <property name="digitCount">
        <number>4</number>
       </property>
       <property name="segmentStyle">
        <enum>QLCDNumber::Flat</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_3">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Mass</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_5">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Damping constant</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QLabel" name="label_8">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>kg</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLCDNumber" name="lcd_damping_constant">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="digitCount">
        <number>4</number>
       </property>
       <property name="segmentStyle">
        <enum>QLCDNumber::Flat</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QDoubleSpinBox" name="box1"/>
     </item>
     <item row="3" column="2">
      <widget class="QDoubleSpinBox" name="box3"/>
     </item>
     <item row="2" column="2">
      <widget class="QDoubleSpinBox" name="box2"/>
     </item>
     <item row="4" column="2">
      <widget class="QDoubleSpinBox" name="box4"/>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>20</y>
      <width>201</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>24</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Spring details</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>530</y>
      <width>201</width>
      <height>91</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QPushButton" name="button1">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Go</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="button2">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="button3">
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1107</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">openglwidget.h</header>
   <container>1</container>
   <slots>
    <signal>advanceFrame()</signal>
    <slot>button_go()</slot>
    <slot>button_stop()</slot>
    <slot>button_reset()</slot>
    <slot>box_dampingConstant()</slot>
    <slot>box_restLength()</slot>
    <slot>box_mass()</slot>
    <slot>box_springConstant()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button1</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>button_go()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>929</x>
     <y>593</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button2</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>button_stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1031</x>
     <y>593</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button3</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>button_reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>929</x>
     <y>629</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>box1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>box_dampingConstant(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>966</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>553</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>box2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>box_restLength(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>966</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>553</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>box3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>box_mass(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>966</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>553</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>box4</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>box_springConstant(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>966</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>553</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>button_go()</slot>
  <slot>button_stop()</slot>
  <slot>box_dampingConstant(double)</slot>
  <slot>box_restLength(double)</slot>
  <slot>box_mass(double)</slot>
  <slot>box_springConstant(double)</slot>
 </slots>
</ui>
